apps/web/pages/api/auth/verify-email.ts

```diff
@@ -0,0 +1,48 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import { z } from "zod";
+
+import dayjs from "@calcom/dayjs";
+import { WEBAPP_URL } from "@calcom/lib/constants";
+import { prisma } from "@calcom/prisma";
+
+const verifySchema = z.object({
+  token: z.string(),
+});
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  const { token } = verifySchema.parse(req.query);
+
+  const foundToken = await prisma.verificationToken.findFirst({
+    where: {
+      token,
+    },
+  });
+
+  if (!foundToken) {
+    res.status(401).json({ message: "No token found" });
```

Comment:
```suggestion
    return res.status(401).json({ message: "No token found" });
```

Reply:
@sean-brydon why was my suggestion ignored? We do need a return here. If there isn't the rest of the code will still run after the response which WILL cause unintended effects.